schema/encoding/jsonschema/schema.go

```diff
@@ -18,132 +13,104 @@ var (
 	ErrNotImplemented = errors.New("not implemented")
 )
 
-// encodeSchema writes JSON Schema keys and values based on s, without the outer curly braces, to w.
-func encodeSchema(w io.Writer, s *schema.Schema) (err error) {
+const (
+	// schemaSizeHint is the maximum number of keys that can be added to a schema map based on schema.Schema.
+	schemaSizeHint = 7
+	// fieldSizeHint is the maximum number of keys that can be added to a schema map based on schema.Field
```

Comment:
It was a hard comment to write, which is never a good sign I guess.

It's not the minimum, but the minimum maximum in a way. The minimum number of fields is 0. This number is the maximum number of properties that can be added directly from the `Field` struct, excluding the filed.Validator attribute, which is what I am trying to state...

I will see if there is a better way.

> I'm also not sure how much value we get out of the specifying this hint, as many of the fields we may add are optional. The default allocator size is probably sufficient.

I will try without a size hints, and then rerun the benchmarks:-)